### ISSUE TITLE
Large discrepancy between test results and ground truth on CAMO dataset

### ISSUE THREAD

**Issue by MarkonH** [WZH0120/SAM2-UNet#13]
Hi,

Thank you for sharing your code. I trained and tested the model using the CAMO dataset, following the exact settings mentioned in your paper (AdamW optimizer, initial learning rate of 0.001, 352 × 352 resolution, batch size of 12, and 50 training epochs). The only modification I made was to test.py to output 0/255.
`for i in range(test_loader.size):
    with torch.no_grad():
        image, gt, name = test_loader.load_data()
        gt = np.asarray(gt, np.float32)
        image = image.to(device)
        res, _, _ = model(image)
        res = torch.sigmoid(res)
        res = F.upsample(res, size=gt.shape, mode='bilinear', align_corners=False)
        res = res.data.cpu().numpy().squeeze()
        
        threshold = 0.5
        res = 255 * (res > threshold).astype(np.uint8)
        
        print("Saving " + name)
        imageio.imsave(os.path.join(args.save_path, name[:-4] + ".png"), res)`
However, I noticed that the test results show a significant discrepancy compared to the ground truth. Could you please clarify if there are any additional steps or special processing that I might be missing?
<img width="659" alt="截屏2024-09-13 下午8 07 00" src="https://github.com/user-attachments/assets/7c98a67a-8274-46f3-ae26-b591ae437848">
<img width="622" alt="截屏2024-09-13 下午8 08 06" src="https://github.com/user-attachments/assets/8a499bef-7cad-4ab2-9458-7eb29328c231">

Thanks in advance!

**Comment by xiongxyowo** [comment]
Hi, we follow the existing camouflage object detection benchmark, where the training set contains 4040 images, 1000 from CAMO and 3040 from COD10K. Is your setup consistent with this? Training on CAMO only may lead to sub-optimal results.

**Comment by MarkonH** [comment]
I see, I mistakenly thought CAMO and COD10K were used as separate training sets. Thank you very much for your clarification!
